ruby/enumerable.md

```diff
@@ -0,0 +1,23 @@
+# Enumerable tricks
+
+## Enumerable#each_cons
+
+Sometimes you need to compare a value in an array with the previous
+value, the next value, or both. [`Enumerable#each_cons`][each_cons docs] to the rescue!
+
+```ruby
+(1..10).each_cons(3) { |element| p element }
```

Comment:
Since you mentioned comparison, how about showing an example with three block parameters and comparing those values?